src/avro_ocf.erl

```diff
@@ -177,6 +195,11 @@ make_block(Header, Objects) ->
   , Header#header.sync
   ].
 
+-spec make_decoder_opts(codec_opts()) -> decoder_opts().
+make_decoder_opts(Options) ->
+  DefaultOptions = [{hook, ?DEFAULT_DECODER_HOOK}],
```

Comment:
maybe create a map here, and use `maps:merger/2`.

Reply:
Yes, I think for now it should be simplified and not copy-pasted as I did, will change

Reply:
I've changed `proplist` to `map` also in the spectype